base/jenkins/jcasc_yamls/02-security.yaml

```diff
@@ -1,12 +1,75 @@
+# - Emergency file provides local admin authentication fallback
+#
 jenkins:
+  # Purpose: CSRF protection to prevent cross-site request forgery attacks
+  # Context: Excludes client IP from crumb to support load balancers and proxies
+  crumbIssuer:
+    standard:
+      excludeClientIPFromCrumb: false
+
+  # Purpose: Disable "Remember me" functionality for enhanced security
+  # Context: Forces users to re-authenticate after session expires
+  # Security: Reduces risk of persistent unauthorized access
+  disableRememberMe: true
+
+  # Purpose: SAML SSO authentication via Linux Foundation identity provider
+  # Context: Integrates with LF SSO for centralized user management
+  # Dependencies: SAML_METADATA_URL and SAML_LOGOUT_URL environment variables
   securityRealm:
-    local:
-      allowsSignup: false
-      users:
-        - id: "admin"
-          password: '${JCASC_ADMIN_PASSWORD}'
+    saml:
+      # SAML binding method for authentication requests
+      binding: "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect"
+      # User attribute mappings from LF SSO SAML claims
+      displayNameAttributeName: "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name"
+      emailAttributeName: "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress"
+      groupsAttributeName: "http://schemas.xmlsoap.org/claims/Group"
+      # Identity provider metadata configuration
+      idpMetadataConfiguration:
+        period: 60  # Metadata refresh interval in minutes
+        url: "${SAML_METADATA_URL}"
+      logoutUrl: "${SAML_LOGOUT_URL}"
+      maximumAuthenticationLifetime: 86400  # 24 hours in seconds
+      usernameCaseConversion: "none"  # Preserve username case from SSO
+      usernameAttributeName: "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier"
+
+  # Purpose: Project-based authorization matrix for role-based access control
+  # Context: Defines permissions for different user groups and roles
+  # Security: Implements principle of least privilege access
   authorizationStrategy:
-    loggedInUsersCanDoAnything:
-      allowAnonymousRead: false
+    projectMatrix:
+      permissions:
+        # Purpose: Administrative access for known Linux Foundation staff
+        # Context: Full Jenkins administration capabilities
+        # Users: lf-jenkins (service account), valderrv, rdetjens (LF staff)
+        - "Overall/Administer:lf-jenkins"
+        - "Overall/Administer:valderrv"
+        - "Overall/Administer:rdetjens"
+
+        # Purpose: Standard permissions for authenticated SAML users
+        # Context: Basic read and build permissions for all authenticated users
+        # Scope: Limited to essential operations, no administrative access
+        - "Overall/Read:authenticated"
+        - "Job/Build:authenticated"
+        - "Job/Cancel:authenticated"
+        - "Job/Read:authenticated"
+        - "Job/Workspace:authenticated"
+        - "View/Read:authenticated"
```

Comment:
IIRC the JCasC for SAML requires you to specify if it's a user or a group in the permissions mapping. Not doing this causes weird issues.

Additionally, lf-releng and lf-sysops groups should be granted full admin permissions permissions